borsar/freq.py

```diff
@@ -188,9 +197,18 @@ def compute_psd(inst, tmin=None, tmax=None, winlen=None, step=None, padto=None,
     if isinstance(inst, mne.BaseEpochs):
         n_per_seg, n_overlap, n_fft = _psd_welch_input_seconds_to_samples(
             inst, winlen, step, padto)
-        psd, freq = psd_welch(inst, tmin=tmin, tmax=tmax, n_fft=n_fft,
-                              picks=picks, n_per_seg=n_per_seg,
-                              n_overlap=n_overlap)
+
+        if new_psd:
+            args = dict(n_fft=n_fft, n_overlap=n_overlap, n_per_seg=n_per_seg)
+            spectrum = inst.compute_psd(method='welch', tmin=tmin, tmax=tmax,
+                                        picks=picks, method_kw=args)
```

Comment:
`method_kw` is a `**` argument; you can either do this:
```suggestion
                                        picks=picks, **args)
```

Or you can just include `n_fft=n_fft, n_overlap=n_overlap, n_per_seg=n_per_seg` directly in the `compute_psd` call here.

Reply:
Thanks! I was too lazy to test this locally ;)